File: src/lib/options.ts
import { parseVariable } from "./helpers.ts"
import { Options } from "./types.ts"

export function parseOptions(lines?: string[]): Options {
  const options = new Map<string, string>()

  for (const line of (lines ?? [])) {
    const v = parseVariable(line)
    if (v != null) {
      options.set(v.name, v.value)
    }
  }

  return {
    layoutFn: options.get("layoutFn") ?? "LAYOUT",
  }
}

File: internal/parser/options.go
package parser

import (
	"strings"

	"github.com/elentok/qmkmd/internal/layout"
)

func ParseOptions(lines []string) layout.Options {
	options := map[string]string{}

	for _, line := range lines {
		name, value, ok := layout.ParseVariable(line)
		if !ok {
			continue
		}
		options[name] = value
	}

	parsed := layout.Options{
		LayoutFn: "LAYOUT",
	}
	if layoutFn, ok := options["layoutFn"]; ok {
		parsed.LayoutFn = layoutFn
	}
	if imports, ok := options["imports"]; ok {
		parts := strings.Split(imports, ",")
		parsed.Imports = make([]string, 0, len(parts))
		for _, part := range parts {
			parsed.Imports = append(parsed.Imports, strings.TrimSpace(part))
		}
	}

	return parsed
}
